### ISSUE TITLE
Constructing a DateTime with an empty string generates an exception

### ISSUE THREAD

**Issue by stuartmcgillcap** [slope-it/clock-mock#31]
`ClockMock` (v0.3.4) generates a `TypeError: DateTime::setTimestamp(): Argument #1 ($timestamp) must be of type int, bool given` when the code tries to construct a `DateTime` with an empty string. This in turn comes from line 34 of the `DateTimeMock` constructor:

```php
$this->setTimestamp(strtotime($datetime, ClockMock::getFrozenDateTime()->getTimestamp()));
```

This behaviour differs from that of the real `DateTime`, which happily returns an object set to the current date and time i.e. equivalent to `new DateTime()`.

Although it would of course be better not to pass in an empty string, this can be hard to avoid in a large legacy application which implictily relies on this (mis)behaviour.

The following test demonstrates the difference between the real `DateTime` and `DateTimeMock`:

```php
<?php

declare(strict_types=1);

namespace ClockMockTests;

use DateTime;
use PHPUnit\Framework\TestCase;
use SlopeIt\ClockMock\ClockMock;

class ClockMockEmptyStringConstructorTest extends TestCase
{
    /** @test */
    public function realDateTimeWithEmptyString(): void
    {
        $this->assertNotNull(new DateTime(''));
    }

    /** @test */
    public function clockMockDateTimeWithEmptyString(): void
    {
        ClockMock::freeze(new DateTime('2023-02-08'));

        $this->expectExceptionMessage(
            'DateTime::setTimestamp(): Argument #1 ($timestamp) must be of type int, bool given'
        );
        new DateTime('');
    }
}

```

**Comment by stuartmcgillcap** [comment]
Using the Elvis operator instead of null coalesce would solve this particular problem, but might have knock-on effects e.g. on zeroes, I'm not sure:

`$datetime = $datetime ?: 'now';`

**Comment by stuartmcgillcap** [comment]
Using the Elvis operator in the constructor instead of `??` would solve this particular problem, but might have knock-on effects e.g. on zeroes, I'm not sure:

`$datetime = $datetime ?: 'now';`

**Comment by asprega** [comment]
Hi @stuartmcgillcap, thanks for your report! I just released version 0.3.5 which fixes the issue: https://github.com/slope-it/clock-mock/releases/tag/0.3.5

Have a good day!

**Comment by stuartmcgillcap** [comment]
Wow that's a quick response - the fix works perfectly. Thank you :)
